code401challenges/src/main/java/code401challenges/mergeSort/MergeSort.java

```diff
@@ -0,0 +1,56 @@
+package code401challenges.mergeSort;
+
+public class MergeSort {
+    public static void sort(int[] arrIn, int left, int right){
+
+        if(right > 1){
+            int mid = (right+1)/2;
+
+            sort(arrIn, 1, mid);
+            sort(arrIn, mid+1, right);
+            merge(arrIn, 1, mid, right);
```

Comment:
sort(arrIn, 1, mid); -> int[] leftArr = sort(<left half>)
sort(arrIn, mid+1, right); -> int[] rightArr = sort(<right half>)
merge(arrIn, 1, mid, right); -> return merge(leftArr, rightArr)

If we change sort to return an array, we can expect it to return a sorted array when called on the left subarray and the right subarray, lets save them into variables.
Merge's job is to merge two arrays, we can simplify the function signature of merge to just take in two arrays. Let it return an array as well
Your solution requires more work to get to an in place solution an I would suggest first building it like this and then think of how to refactor it to be in place